### ISSUE TITLE
Setting an email address as requester limits entities choice

### ISSUE THREAD

**Issue by felag** [glpi-project/glpi#15625]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.9

### Bug description

When creating a new ticket, if I enter an email address as requester (**not** using a GLPI users account), page is updated with limitations on entities I can select.

(In my case there were only 3 entities.)

### Relevant log output

```shell
None, but enabling debug mode indicated that the limited 3 entities where given by this SQL query:


SELECT DISTINCT `entities_id`, `is_recursive`  FROM `glpi_profiles_users`  WHERE `users_id` = '0';
```

### Page URL

https://[GLPI_SERVER_URL]/front/ticket.form.php

### Steps To reproduce

1. Login to GLPI
2. Create a new ticket
3. enter an email address as requester (on middle right of page)
4. select such email address to update requester field (page is updated)
5. try to change entity of ticket (on top right of page)
6. Not all entities are selectable

### Anything else?

It appeared that content of table glpi_profiles_users only showed 3 lines with users_id = 0.
After deleting such 3 lines in database, problem disappeared (*edit: I mean: it became possible to select any entity again*).

Our database is used since a long time and I have no idea where this came from.

I thought it was worth mentioning: maybe can a warning be displayed during update to indicate such table content ?

**Comment by cedric-anne** [comment]
Hi,

It should be fixed by #15626.

**Comment by felag** [comment]
@cedric-anne Thanks for your prompt reply.

I'll let you know when it is corrected.
